lib/handlers/quote/injector.ts

```diff
@@ -91,6 +91,9 @@ export class QuoteHandlerInjector extends InjectorSOR<
     let router
     switch (algorithm) {
       case 'legacy':
+        if (!v3QuoteProvider) {
+          throw new Error('V3 Quote Provider not set for Legacy Routings')
+        }
```

Comment:
lgtm, only thing is im not sure if this is the behavior we want for legacy router, because essentially it would break legacy router for all chains except those specifically defined here (all except Optimism and Arb)

Reply:
what do you suggest instead

Reply:
I think if the v3QuoteProvider is not set here, Legacy Router would break anyway since it doesn't have a fallback quote provider right?

Reply:
Yeah it doesn't have a fallback and expects it to be passed into the constructor. 

If the quoteProvider is undefined, we could create the quoteProvider here with the default settings and just pass into legacy router only, which would mimic the previous behavior. Since I assume we would want legacy to still work

Reply:
Agree with Eric, lets put the old behavior here for legacy